eipw-lint/tests/lint_markdown_link_eip.rs

```diff
@@ -0,0 +1,190 @@
+       .check_slice(None, src)
+       .run()
+       .await
+       .unwrap()
+       .into_inner();
+    assert_eq!(
+        reports,
+        r#"error[markdown-link-eip]: link text does not match link destination
+  |
+4 | [EIP-1: Foo](./eip-1.md)
+  |
+  = info: link text should match `[EIP|ERC-1]`
+"#
+    );
+}
+
+#[tokio::test]
+async fn link_text_missing_eip() {
+    let src = r#"---
+header: value1
+---
+[Another Proposal](./eip-1.md)
+"#;
+
+    let reports = Linter::<Text<String>>::default()
+       .clear_lints()
+       .deny("markdown-link-eip", LinkEip(r"eip-([^.]*)\.md(#.+)?$".to_string()))
+       .check_slice(None, src)
+       .run()
+       .await
+       .unwrap()
+       .into_inner();
+    assert_eq!(
+        reports,
+        r#"error[markdown-link-eip]: link text does not match link destination
+  |
+4 | [Another Proposal](./eip-1.md)
+  |
+  = info: link text should match `[EIP|ERC-1]`
+"#
+    );
+}
+
+#[tokio::test]
+async fn link_text_missing_section_description() {
+    let src = r#"---
+header: value1
+---
+[EIP-1](./eip-1.md#motivation)
+"#;
+
+    let reports = Linter::<Text<String>>::default()
+       .clear_lints()
+       .deny("markdown-link-eip", LinkEip(r"eip-([^.]*)\.md(#.+)?$".to_string()))
+       .check_slice(None, src)
+       .run()
+       .await
+       .unwrap()
+       .into_inner();
+    assert_eq!(
+        reports,
+        r#"error[markdown-link-eip]: link text does not match link destination
+  |
+4 | [EIP-1](./eip-1.md#motivation)
+  |
+  = info: link text should match `[EIP|ERC-1<section-description>]`
+"#
+    );
+}
+
+#[tokio::test]
+async fn eip_number_mismatch_with_section() {
+    let src = r#"---
+header: value1
+---
+[EIP-2: Hello](./eip-1.md#abstract)
+"#;
+
+    let reports = Linter::<Text<String>>::default()
+       .clear_lints()
+       .deny("markdown-link-eip", LinkEip(r"eip-([^.]*)\.md(#.+)?$".to_string()))
+       .check_slice(None, src)
+       .run()
+       .await
+       .unwrap()
+       .into_inner();
+    assert_eq!(
+        reports,
+        r#"error[markdown-link-eip]: link text does not match link destination
+  |
+4 | [EIP-2: Hello](./eip-1.md#abstract)
+  |
+  = info: link text should match `[EIP|ERC-1<section-description>]`
+"#
+    );
+}
+
+#[tokio::test]
+async fn link_text_missing_eip_with_section() {
+    let src = r#"---
+header: value1
+---
+[Another Proposal](./eip-1.md#rationale)
+"#;
+
+    let reports = Linter::<Text<String>>::default()
+       .clear_lints()
+       .deny("markdown-link-eip", LinkEip(r"eip-([^.]*)\.md(#.+)?$".to_string()))
+       .check_slice(None, src)
+       .run()
+       .await
+       .unwrap()
+       .into_inner();
+    assert_eq!(
+        reports,
+        r#"error[markdown-link-eip]: link text does not match link destination
+  |
+4 | [Another Proposal](./eip-1.md#rationale)
+  |
+  = info: link text should match `[EIP|ERC-1<section-description>]`
```

Comment:
Similar kind of comment here. The help text should (if possible) include the correct text to use. Authors might not understand how to make something "match".